base/sysinfo.jl

```diff
@@ -526,4 +526,50 @@ function which(program_name::String)
 end
 which(program_name::AbstractString) = which(String(program_name))
 
+### Submodule with information about CPU features
+module CPUID
+
+"""
+    ISA(features::Set{String})
+
+A structure which represents the Instruction Set Architecture (ISA) of a
+computer.  It holds the `Set` of features of the CPU.
+"""
+struct ISA
+    features::Set{UInt32}
+end
+
+Base.:<=(a::ISA, b::ISA) = a.features <= b.features
+
+include("features_h.jl")
+
+# Sets of features for the x86-64 microarchitectures come from
+# https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html
+const X86_64_GENERIC = ISA(Set{UInt32}()) # Missing: mmx, sse, sse2
```

Comment:
[`src/features_x86.h`](https://github.com/JuliaLang/julia/blob/08486888baad7584fe25ac4c6aee97730890968c/src/features_x86.h) doesn't have `mmx`, `sse`, nor `sse2`, do we just assume they're always there in x86-64 chips (which should be a safe assumption)?

Reply:
Yes those are required